Formula/jolie.rb

```diff
@@ -0,0 +1,21 @@
+class Jolie < Formula
+  desc "The Jolie Language Interpreter"
+  homepage "https://www.jolie-lang.org/"
+  url "https://github.com/jolie/jolie/releases/download/v1.9.0/jolie-1.9.0.jar"
+  sha256 "1510ed7f114909eb79670462571ab0734a5b01e57d26da6fd1cf9ef6c67eff6e"
+
+  depends_on "openjdk"
+
+  def install
+    system "java", "-jar", "jolie-#{version}.jar", "-jh", libexec, "-jl", bin
+    Dir.each_child(bin) do |filename|
+      inreplace "#{bin}/#{filename}" do |s|
+        s.gsub! /\$JOLIE_HOME/, libexec
+      end
+    end
+  end
+
+  test do
+    system "#{bin}/jolie", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Meaningful test included